docs2/site/docs/getting-started/errors.md

```diff
@@ -1,78 +1,203 @@
 # Error Handling
 
-The `ExecutionResult` provides an `Errors` property which includes any errors encountered
-during execution. Errors are returned [according to the spec](https://graphql.github.io/graphql-spec/June2018/#sec-Errors),
-which means stack traces are excluded. The `ExecutionResult` is transformed to what the spec
-requires using one or the another `IDocumentWriter`.
+Errors within the GraphQL engine can be thought of as falling into one of three groups: schema errors,
+input errors and processing errors. For instance, if an invalid query was passed to the `DocumentExecuter`,
+it would be considered an input error, and a `SyntaxError` would be thrown. Or if an invalid enum string
+was passed as a variable to a query, an `InvalidValueError` would be thrown. All validation rules that
+fail their respective tests are treated as input errors.  Processing errors typically represent unanticipated
+exceptions occurring during the execution of a field resolver, such as a timeout during a database operation.
 
-For example `GraphQL.NewtonsoftJson.DocumentWriter` uses [JSON.NET](https://www.nuget.org/packages/Newtonsoft.Json)
-and `GraphQL.SystemTextJson.DocumentWriter` uses new .NET Core memory optimized serializer from
-[`System.Text.Json`](https://docs.microsoft.com/en-us/dotnet/api/system.text.json). For JSON.NET you can change
-what information is provided by setting your values (`ContractResolver`, Converters`, etc.)
-in `JsonSerializerSettings` passed to `DocumentWriter` constructor. For `System.Text.Json`
-serializer you can configure `JsonSerializerOptions` passed to `DocumentWriter` constructor.
```

Comment:
Do I understand correctly that you specifically excluded this information because it is more related to the serialization of the entire result and not just errors? We won't mention serializers on this documentation page?

Reply:
Basically.  I just couldn't see how it related to errors.  I had a sentence something like this written, but it just didn't really fit anywhere:

> Implementations of `IDocumentWriter` are available for two json serialization packages, `System.Text.Json` and `Newtonsoft.Json`.  Both provide a constructor that accepts a `IErrorInfoProvider` ....

Just didn't seem to be relevant, and it didn't flow well with the rest of the paragraph.  Talking about contract resolvers is way more deep than a guide on error handling, I figured.  Especially since our new `IErrorInfoProvider` takes the place of customizing the `IDocumentWriter` or its inner workings.

Reply:
(as it relates to error handling)

Reply:
Agreed